x-pack/plugins/maps/common/constants.ts

```diff
@@ -99,6 +99,10 @@ export enum ES_GEO_FIELD_TYPE {
   GEO_SHAPE = 'geo_shape',
 }
 
+// Using strings instead of ES_GEO_FIELD_TYPE enum to avoid typeing errors where IFieldType.type is compared to value
+// TODO replace ES_GEO_FIELD_TYPE enum with strings from core kibana.
```

Comment:
fwiw, I would just remove the todo-comment. We have around 30 ones already in Maps, most just gathering dust. If it's worth following up on explicitly, let's create an issue.